raiden/blockchain/events.py

```diff
@@ -245,13 +249,41 @@ class BlockchainEvents:
     def __init__(self, chain_id: ChainID):
         self.chain_id = chain_id
         self.event_listeners: List[EventListener] = list()
+        self.last_sync_log_time: Optional[datetime] = None
+        self.last_sync_log_block: BlockNumber = BlockNumber(0)
+
+    def _log_sync_progress(self, from_block: BlockNumber, to_block: BlockNumber) -> None:
+        """
+        In case we have fallen far behind with synchronizing blockchain events,
+        display a sync progress message every few second.
+        """
+        blocks_to_sync = to_block - from_block
+        if blocks_to_sync > 100:
+            if self.last_sync_log_time is None:
+                self.last_sync_log_time = datetime.now()
+                self.last_sync_log_block = from_block
+                log.info("Synchronizing blockchain events", blocks_left=blocks_to_sync)
+            else:
+                now = datetime.now()
+                elapsed = (now - self.last_sync_log_time).total_seconds()
+                if elapsed >= 5.0:
+                    blocks_per_second = (from_block - self.last_sync_log_block) / elapsed
+                    log.info(
+                        "Synchronizing blockchain events",
+                        blocks_left=blocks_to_sync,
+                        blocks_per_second=blocks_per_second,
+                    )
+                    self.last_sync_log_time = now
+                    self.last_sync_log_block = from_block
 
     def poll_blockchain_events(self, block_number: BlockNumber) -> Iterable[DecodedEvent]:
         """ Poll for new blockchain events up to `block_number`. """
 
         for event_listener in self.event_listeners:
             assert isinstance(event_listener.filter, StatelessFilter)
 
+            self._log_sync_progress(event_listener.filter.from_block_number(), block_number)
```

Comment:
We do this for each `event_listener`, but the listener is not logged. This will make the output irritating in some cases:

> Synchronizing blockchain events, blocks_left=200
> Synchronizing blockchain events, blocks_left=80, blocks_per_second=24
> Synchronizing blockchain events, blocks_left=200
> Synchronizing blockchain events, blocks_left=82, blocks_per_second=23.9

This would be less surprising if we also log the name (or another identifier) for the listener.

Reply:
Thanks for pointing this out. I moved the logging out of the loop instead, we actually don't want to log for each listener seperately...